feat(rust): add examples section to ockam message send command's help

chrischiedo opened the pull request:
**PR purpose:** Address issue #3257 

## Current Behaviour
`ockam message send --help`

does not display any examples.

## Desired Behaviour
As part of the help, the following examples section should be shown:
```
EXAMPLES

    # Create two nodes
    > ockam node create n1
    > ockam node create n2

    # Send a message to the uppercase service on node n1
    > ockam message send hello --to /node/n1/service/uppercase
    HELLO

    # Send a message to the uppercase service on node n1 from node n1
    > ockam message send hello --from /node/n2 --to /node/n1/service/uppercase
    HELLO

    # Create a secure channel from node n1 to the api service on node n1
    # Send a message through this encrypted channel to the uppercase service
    > ockam secure-channel create --from /node/n1 --to /node/n2/service/api \
        | ockam message send hello --from /node/n1 --to -/service/uppercase
    HELLO

LEARN MORE
```

chrischiedo committed:
docs(rust): add examples section to the help for `ockam message send` command

mrinalwadhwa commented on implementations/rust/ockam/ockam_command/src/message/send.rs:
```diff
@@ -13,7 +13,32 @@ use ockam_multiaddr::MultiAddr;
+    $ ockam message send hello --from /node/n2 --to /node/n1/service/uppercase
+    HELLO
+
+    # Create a secure channel from node n1 to the api service on node n1
```
```suggestion
    # Create a secure channel from node n1 to the api service on node n2
```

mrinalwadhwa reviewed (dismissed):
@chrischiedo thank you for spending time on this. I left a minor suggestion

mrinalwadhwa reviewed (dismissed):
That looks great. Thank you!

mrinalwadhwa commented:
@chrischiedo one thing to note:

1. Github thinks your commit is Unverified.
<img width="916" alt="Screen Shot 2022-08-21 at 3 21 44 PM" src="https://user-images.githubusercontent.com/159583/185813244-3dd5c6f4-3c8b-4685-a26d-d6d5045d214f.png">
 
This is because your git author email is set to `chiedo1986ouma7*`
This is also why our commit check is failing.

You can fix this by [setting](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address):

```
git config --global user.email "email@example.com"
```

This email must be an email github knows about you and the email you mention in our [contributors](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file

After this create the commit again.

chrischiedo commented:
> @chrischiedo one thing to note:
> 
> 1. Github thinks your commit is Unverified.
> 
> <img alt="Screen Shot 2022-08-21 at 3 21 44 PM" width="916" src="https://user-images.githubusercontent.com/159583/185813244-3dd5c6f4-3c8b-4685-a26d-d6d5045d214f.png">
> 
> This is because your git author email is set to `chiedo1986ouma7*` This is also why our commit check is failing.
> 
> You can fix this by [setting](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address):
> 
> ```
> git config --global user.email "email@example.com"
> ```
> 
> This email must be an email github knows about you and the email you mention in our [contributors](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file
> 
> After this create the commit again.

I see. But I wonder why my other PR for adding my details to the contributors file didn't raise any issues...
Let me reset my email config and try again.

mrinalwadhwa commented:
Maybe because you created it in the browser .. it uses the github provided email address instead of your main email.

mrinalwadhwa commented:
If you need help, I can amend the commit for you before merging ... let me know.

chrischiedo commented:
> If you need help, I can amend the commit for you before merging ... let me know.

I've fixed the issue with my email.

mrinalwadhwa commented:
@chrischiedo this looks ready to merge. I squashed all the commits into a single one with `git rebase -i` and amended the commit message a bit. I will merge this shortly.

Congratulations on an awesome first contribution 🥳 🎈

chrischiedo commented:
@mrinalwadhwa, Thank you very much for your quick response! This being my first contribution, you can't imagine how excited I'm!!
I assume that I can go ahead and work on the other (closely related) issues? (#3265 and #3266 ). If yes, I should be done with those by the end of tomorrow (Monday).

mrinalwadhwa commented:
@chrischiedo that is awesome 🎉 

I've been writing a long help description for secure channel and secure channel listener today so I think that will close https://github.com/build-trust/ockam/issues/3265 and https://github.com/build-trust/ockam/issues/3266 ... however I will create more similar issues later today and leave a message here. So please have a look at one of those tomorrow.

chrischiedo commented:
> @chrischiedo that is awesome 🎉
> 
> I've been writing a long help description for secure channel and secure channel listener today so I think that will close #3265 and #3266 ... however I will create more similar issues later today and leave a message here. So please have a look at one of those tomorrow.

@mrinalwadhwa, Okay. This is noted. Once you're done creating the new issues, I'll take a look at them and proceed accordingly.

In the meantime...
(I know I should probably share this in the _discussions_ section, but please bear with me)
Are there any _other_ specific open issues that you'd recommend I look at? FYI: I consider myself an "advanced beginner" in Rust, and I'm relatively comfortable with building CLI apps using clap.
I know you have created this [_discussion_](https://github.com/build-trust/ockam/discussions/3206) thread for good first issues but I was wondering if you'd perhaps have a great suggestion in mind...

Once again thanks a lot for your help and guidance. Much appreciated.

mrinalwadhwa commented:
@chrischiedo I'm super glad you're excited about contributing so don't hesitate to ask questions. Anywhere is fine as long as I don't miss seeing the :)

> I consider myself an "advanced beginner" in Rust, and I'm relatively comfortable with building CLI apps using clap.

Based on that context. Here's a suggestion:

We are in the middle of refactoring various commands to have a simpler structure. Have a look at how the [`secure-channel create`](https://github.com/build-trust/ockam/pull/3293/files#diff-2934a64b2c22d750f6278bf1a1787880035359aef74ab6bf6cabc18a8030d6bd) and [`secure-channel delete`](https://github.com/build-trust/ockam/pull/3293/files#diff-a5072342343bd9f300be720e7a4dbe7d8f5c43a3dc3f8974a509d58b6310c41a) commands are change in PR #3293 which is currently a draft.

If you're interested, please do the same refactor to use the Rpc helpers for all the commands in the tcp folder https://github.com/build-trust/ockam/tree/develop/implementations/rust/ockam/ockam_command/src/tcp

One command at a time, in each PR, maybe the best approach so we have an opportunity to give feedback early.

Let me know if/when you'd like to work on this.

chrischiedo commented:
Hey @mrinalwadhwa,

I've glanced at the changes made in PR #3293 and I think it needs a bit of a deep dive (at least to understand how the different modules interact). Let me try to grok it and then get back to you by tomorrow. I may also seek clarification on things that might not be clear to me...

Also, by "Rpc helpers" you mean the rpc associated (helper) functions/methods defined within the _util_ directory, right?

mrinalwadhwa commented:
>  "Rpc helpers" you mean the rpc associated (helper) functions/methods defined within the util directory, right?

Yes I do.
Please ask any questions you may have along the way.

mrinalwadhwa commented:
@chrischiedo since you were looking for these ... I added some more example help improvement issues yesterday https://github.com/build-trust/ockam/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22

chrischiedo commented:
> @chrischiedo since you were looking for these ... I added some more example help improvement issues yesterday https://github.com/build-trust/ockam/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22

Hey @mrinalwadhwa,

I've made a [PR](https://github.com/build-trust/ockam/pull/3350) addressing the 3 help improvement issues.

I noted one thing though with the `ockam forwarder create` command: You didn't leave any comments on the examples section, and therefore I added some of the comments myself. Please let me know if my comments are accurate.
